test/output_queue_tests/output_queue_test.py

```diff
@@ -1,49 +1,110 @@
+import queue
+import time
+import pytest
+
+import mido
 from src.output_queue.output_queue import OutputQueue
+from src.common.midi_event import MidiEvent
 
-class TestIsEven:
-    def test_even(self):
-        component = OutputQueue()
-        evenVal = 2
-
-        actual = component.is_even(evenVal)
-        expected = True
-        
-        assert actual is expected
-
-    def test_odd(self):
-        component = OutputQueue()
-        evenVal = 1
-        
-        actual = component.is_even(evenVal)
-        expected = False
-        
-        assert actual is expected
-
-class TestInRange:
-    def test_lower_bound(self):
-        component = OutputQueue()
-        lower_val = 3
-
-        actual = component.in_range(lower_val)
-        expected = False
-        
-        assert actual is expected
-    
-    def test_upper_bound(self):
-        component = OutputQueue()
-        upper_val = 8
-
-        actual = component.in_range(upper_val)
-        expected = False
-        
-        assert actual is expected
+FAR_FUTURE_TIMESTAMP = 1e10
+
+class DummyPort:
+    def __init__(self):
+        self.name = "DummyPort"
+        self.sent_messages = []
+
+    def close(self):
+        pass
+
+    def send(self, msg):
+        self.sent_messages += [msg]
+
+@pytest.fixture
+def output_queue(mocker):
+    output_queue = OutputQueue(queue.PriorityQueue())
+    dummy_port = DummyPort()
+
+    mock_get_output_names = mocker.patch("mido.get_output_names")
+    mock_get_output_names.return_value = [dummy_port.name]
+
+    mock_open_output = mocker.patch("mido.open_output")
+    mock_open_output.return_value = dummy_port
+
+    return output_queue
+
+class TestPorts:
+    def test_output_queue(self, mocker, output_queue):
+        test_output_devices = ["Output1", "Output2"]
+
+        mock_get_output_names = mocker.patch("mido.get_output_names")
+        mock_get_output_names.return_value = test_output_devices
+
+        assert test_output_devices == output_queue.get_device_list()
+
+    def test_select_bad_device(self, mocker, output_queue):
+        test_output_devices = ["Output1", "Output2"]
+        bad_device_name = "Output 3"
+
+        mock_get_output_names = mocker.patch("mido.get_output_names")
+        mock_get_output_names.return_value = test_output_devices
+
+        with pytest.raises(Exception):
+            output_queue.select_device(bad_device_name)
+
+class TestRun:
+    @pytest.mark.timeout(0.5)
+    def test_signal_stop(self, output_queue):
+        output_queue.start()
+        output_queue.signal_stop()
+        output_queue.join()
     
-    def test_within_bounds(self):
-        component = OutputQueue()
-        within_range_val = 5
+    def test_empty_queue(self, output_queue):
+        output_queue.select_device()
+
+        assert output_queue._check_queue() == 0
```

Comment:
move this to a variable, it took a sec to figure out this is what you were testing

`actual =output_queue._check_queue()`
`assert actual == 0`